views/lib/components/files-storage/files-grid.hbs

```diff
@@ -0,0 +1,102 @@
+							role="button">
+							<div class="card-title">
+								<div class="file-preview-wrapper col-sm-1 no-padding hidden-xs-down">
+									<div class="file-preview"
+										 style="background-image: url({{this.thumbnail}}); -webkit-background-size: 25px ;background-size: 25px;"></div>
+								</div>
+								<a class="col-sm-10 title">
+									{{this.name}}
+									{{#if this.size}}
+										<small style="float: right;" class="file-info" id="{{this.id}}"></small>
+									{{/if}}
+								</a>
+							</div>
+							<div class="card-text">
+							  <small class="file-location">{{#if this.context}}{{$t "files.text.foundIn"}} {{this.context}}{{/if}}</small><br>
+								{{#ifeq this.securityCheck.status "blocked"}}
+									<button
+									class="file-btn btn-file-danger"
+									title="{{$t "files.label.dangerousCode"}}">
+										<i aria-hidden="true" class="fa fa-exclamation-triangle"></i>
+									</button>
+								{{else}}
+									<button data-href="/api/v3/file/download/{{../this.id}}/{{../this.name}}"
+										class="file-btn"
+										data-method="download"
+										title="{{$t "files.label.downloadFile"}}">
+										<i aria-hidden="true" class="fa fa-cloud-download"></i>
+									</button>
+								{{/ifeq}}
+
+								<button
+								    data-file-id="{{this.id}}"
+									data-file-name="{{this.name}}"
+									data-file-path="{{this.path}}"
+									title="{{$t "files.label.renameFile"}}"
+									class="file-btn fa fa-edit file-name-edit">
+								</button>	
+								{{#userHasPermission "FILE_DELETE"}}
+								<button data-href="/files/file/"
+								   class="file-btn"
+								   data-method="delete"
+								   title="{{$t "files.label.deleteFile"}}"
+								   data-file-id="{{../this.id}}"
+								   data-file-name="{{../this.name}}">
+									<i aria-hidden="true" class="fa fa-trash-o"></i>
+								</button>
+								{{/userHasPermission}}
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		{{/each}}
+	</div>
+</section>
+
+{{#embed "lib/components/modal-form" class="edit-modal" title=($t "files.headline.newFolder") closeLabel=($t "global.button.cancel") submitLabel=($t "global.button.createFolder")}}
+	{{#content "fields"}}
+        <label for="newDirName" hidden>{{$t "files.input.foldername"}}</label>
+		<input id="newDirName" type="text" class="form-control" name="new-dir-name" placeholder="{{$t "files.input.foldername"}}" required />
+	{{/content}}
+{{/embed}}
+
+{{#embed "lib/components/modal" class="delete-modal" title=($t "global.text.areYouSure")}}
+	{{#content "body"}}
+
+		<div class="modal-footer">
+			<button type="button" class="btn btn-secondary btn-close">
+				{{$t "global.button.cancel" }}
+			</button>
+
+			<button type="submit" class="btn btn-primary btn-submit">
+				{{$t "global.headline.delete" }}
+			</button>
+		</div>
+	{{/content}}
+{{/embed}}
+
+{{#embed "lib/components/modal-form" class="rename-modal"}}
```

Comment:
The file combine different embed but the section are is added directly. 
We should outsource it too, or put the others also to the file without embed.